Adopted IntelliJ-standard <caret>-based CodeInsightFixture for testing

viatra2 opened the pull request:
(corrected attribution)

viatra2 committed:
releng: extract path to yaml plugin into gradle.properties

- avoids merge conflicts on pulling routine changes for build.gradle (e.g, version bumps)

viatra2 committed:
tests: codeInsightFixture (<caret>-based) tests for swagger completion

- mediaType in top-level produces/consumes
- enumeration value for parameter/in (failing, disabled)

viatra2 committed:
tests: add the same completion tests for json swagger flow

- parallel json input files, same checks
- TODO: simplify the json vs yaml testing by migration to junit4

viatra2 committed:
tests: codeInsightFixture (<caret>-based) tests for swagger completion

- migrated to junit4 to simplify @Parameterized run
- @Ignore now works, cool

viatra2 committed:
more basic completion tests: http operations for path items

semonte commented on src/test/java/org/zalando/intellij/swagger/completion/CaretCompletionTest.java:
```diff
@@ -0,0 +1,96 @@
+@RunWith(Parameterized.class)
+public class CaretCompletionTest {
+    @Rule
+    public TestName testName = new TestName();
```
This can probably be removed?

viatra2 replied:
Agreed.

semonte commented on src/test/java/org/zalando/intellij/swagger/completion/CaretCompletionTest.java:
```diff
@@ -0,0 +1,96 @@
+    @After
+    public void tearDownAfter() throws Exception {
+        myFixture.tearDown();
+        myFixture = null;
```
This can probably be removed?

viatra2 replied:
Sorry, don't understand. 
- call to tearDown is required. 
- setting field to null is not required but does not look wrong to me

semonte replied:
I meant that `myFixture = null;` can be removed if everything works without that assignment.

semonte commented on src/test/java/org/zalando/intellij/swagger/completion/SwaggerFixture.java:
```diff
@@ -0,0 +1,122 @@
+
+        @NotNull
+        public String getFileName(@NotNull String base) {
+            return base.endsWith(".") ? base + myFileExtension : base + "." + myFileExtension;
```
Why this logic? Can we use

`return base + "." + myFileExtension;`

viatra2 replied:
Agreed.

semonte commented on src/test/java/org/zalando/intellij/swagger/completion/SwaggerFixture.java:
```diff
@@ -0,0 +1,122 @@
+    }
+
+    @NotNull
+    private static String toAbsolutePath(@NotNull String resourcesRelatedPath) {
```
For simplicity I would write this to the following:

```
@NotNull
    private static String toAbsolutePath(@NotNull String resourcesRelatedPath) throws URISyntaxException {
        URL url = ClassLoader.getSystemClassLoader().getResource(resourcesRelatedPath);
        return new File(url.toURI()).getAbsolutePath();
    }
```

viatra2 replied:
Indeed.

semonte commented on src/test/resources/testData/completion/enum_parameters_in.json:
```diff
@@ -0,0 +1,28 @@
+{
```
Please rename testData to testdata, all files and folders are lower cased.

viatra2 replied:
will do.

viatra2 replied:
oh well, changing only the case is a problem for git+idea+mac, never knew that. 
will have to rename `testData.completion` -> `testing.completion`, and `mediaType_*` -> `media_type_*`

semonte commented:
Thank you for the pull requests, looks good for the most parts! I added some comments.

semonte commented on src/test/java/org/zalando/intellij/swagger/completion/SwaggerFixture.java:
```diff
@@ -0,0 +1,122 @@
+        }
+    }
+
+    public static class AssertableList<T> {
```
I think generics are not needed here, we are only playing around with strings.

`AssertableList<T>` to `AssertableList` and
`private final List<T> myActualList;` to `private final List<String> myActualList;`

viatra2 replied:
Agreed, will do.

viatra2 committed:
Addressing comments for PR #12

- all files and folders to have lower-case names
- - testData -> testing (can't rename to testdata, since case-only renaming don't work well)
- - mediaType_* -> media_type_*

viatra2 committed:
Addressing comments for PR #12

- AssertableList only bound to String, generics not needed
- toAbsolutePath simplified
- JsonOrYaml#getFileName simplified
- unused CaretCompletionTest#testName removed

viatra2 commented:
@semonte please revisit, hopefully addressed everything except https://github.com/zalando/intellij-swagger/pull/12#discussion_r64220885

viatra2 committed:
Addressing comments for PR #12

- useless nullification removed for simplicity

viatra2 commented:
@semonte actually addressed a last one as well. thanks

viatra2 committed:
Addressing comments for PR #12

- last UpperCaseFileName fixed (topLevel.* -> top_level.*)
